tests/unit_tests/steps/test_compile_c.py

```diff
@@ -50,110 +59,138 @@ def test_compile_c_wrong_compiler(content):
 
     # Now check that _compile_file detects the incorrect class of the
     # C compiler
-    mp_common_args = mock.Mock(config=config)
-    with pytest.raises(RuntimeError) as err:
-        _compile_file((None, mp_common_args))
-    assert ("Unexpected tool 'mock_c_compiler' of category "
-            "'FORTRAN_COMPILER' instead of CCompiler" in str(err.value))
+    mp_common_args = Mock(config=config)
+    with raises(RuntimeError) as err:
+        _compile_file((Mock(), mp_common_args))
+    assert str(err.value) == "Unexpected tool 'some C compiler' of category " \
```

Comment:
pep8 (https://peps.python.org/pep-0008/#maximum-line-length):
"Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation."

Reply:
Not sure I agree with that but it is the standard.

Reply:
I admit, I didn't like it initially, too - but got used to it.